From: Ben Hutchings <ben@decadent.org.uk>
Date: Sat, 19 Oct 2013 22:37:58 +0100
Subject: lib/raid6: Disable altivec8 algorithm
Forwarded: no

gcc 4.8 generates code for the altivec8 algorithm that requires
the _restvr_20 and _savevr_20 subroutines from the run-time library.
The kernel does not yet include these subroutines.

--- a/lib/raid6/Makefile
+++ b/lib/raid6/Makefile
@@ -4,7 +4,7 @@ raid6_pq-y	+= algos.o recov.o tables.o i
 		   int8.o int16.o int32.o
 
 raid6_pq-$(CONFIG_X86) += recov_ssse3.o recov_avx2.o mmx.o sse1.o sse2.o avx2.o
-raid6_pq-$(CONFIG_ALTIVEC) += altivec1.o altivec2.o altivec4.o altivec8.o
+raid6_pq-$(CONFIG_ALTIVEC) += altivec1.o altivec2.o altivec4.o # altivec8.o
 
 hostprogs-y	+= mktables
 
--- a/lib/raid6/algos.c
+++ b/lib/raid6/algos.c
@@ -64,7 +64,7 @@ const struct raid6_calls * const raid6_a
 	&raid6_altivec1,
 	&raid6_altivec2,
 	&raid6_altivec4,
-	&raid6_altivec8,
+/*	&raid6_altivec8, */
 #endif
 	&raid6_intx1,
 	&raid6_intx2,
